flink-clients/src/main/java/org/apache/flink/client/cli/CliFrontend.java

```diff
@@ -219,34 +219,49 @@ protected void run(String[] args) throws Exception {
 
 		final ProgramOptions programOptions = ProgramOptions.create(commandLine);
 
-		final PackagedProgram program =
-				getPackagedProgram(programOptions);
+		final List<URL> jobJars = getJobJarAndDependencies(programOptions);
 
-		final List<URL> jobJars = program.getJobJarAndDependencies();
 		final Configuration effectiveConfiguration = getEffectiveConfiguration(
 				activeCommandLine, commandLine, programOptions, jobJars);
 
 		LOG.debug("Effective executor configuration: {}", effectiveConfiguration);
 
+		final PackagedProgram program = getPackagedProgram(programOptions, effectiveConfiguration);
+
 		try {
 			executeProgram(effectiveConfiguration, program);
 		} finally {
 			program.deleteExtractedLibraries();
 		}
 	}
 
-	private PackagedProgram getPackagedProgram(ProgramOptions programOptions) throws ProgramInvocationException, CliArgsException {
+	/**
+	 * Get all provided libraries needed to run the program from the ProgramOptions.
+	 */
+	public List<URL> getJobJarAndDependencies(ProgramOptions programOptions) throws FileNotFoundException, ProgramInvocationException {
+		String entryPointClass = programOptions.getEntryPointClassName();
+		String jarFilePath = programOptions.getJarFilePath();
+
+		File jarFile = jarFilePath != null ? getJarFile(jarFilePath) : null;
+
+		return PackagedProgram
+			.getJobJarAndDependencies(jarFile != null ? jarFile : null, entryPointClass);
```

Comment:
This can become `return PackagedProgram.getJobJarAndDependencies(jarFile, entryPointClass);`

Reply:
fixed